python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -379,3 +379,59 @@ def get_logs_for_run(
         cursor=conn.cursor,
         hasMore=conn.has_more,
     )
+
+
+def _fetch_runs_not_in_backfill(
+    instance, cursor: Optional[str], limit: Optional[int]
+) -> Sequence[RunRecord]:
+    if limit is None:
+        new_runs = instance.get_run_records(cursor=cursor)
+        return [run for run in new_runs if run.dagster_run.tags.get(BACKFILL_ID_TAG) is None]
+
+    runs = []
+    while len(runs) < limit:
+        new_runs = instance.get_run_records(limit=limit, cursor=cursor)
+        if len(new_runs) == 0:
+            return runs
+        cursor = new_runs[-1].dagster_run.run_id
+        runs.extend([run for run in new_runs if run.dagster_run.tags.get(BACKFILL_ID_TAG) is None])
+
+    return runs[:limit]
+
+
+def get_mega_runs(
+    graphene_info: "ResolveInfo",
+    cursor: Optional[str] = None,
+    limit: Optional[int] = None,
+) -> Sequence["GrapheneMegaRun"]:
+    """Returns a merged list of backfills and single runs (runs that are not part of a backfill).
+
+    Cursor format: run_id;backfill_id
+    """
+    from ..schema.pipelines.pipeline import GrapheneMegaRun
+
+    check.opt_str_param(cursor, "cursor")
+    check.opt_int_param(limit, "limit")
+
+    instance = graphene_info.context.instance
+
+    run_cursor, backfill_cursor = cursor.split(";") if cursor else (None, None)
+
+    # get_backfills doesn't support many filters. Need to implement some filtering before we can
+    # add that as a param
+    backfills = instance.get_backfills(cursor=backfill_cursor, limit=limit)
+    runs = _fetch_runs_not_in_backfill(instance, cursor=run_cursor, limit=limit)
+
+    # order runs and backfills by create_time. typically we sort by storage id but that won't work here since
+    # they are different tables
+    all_mega_runs = sorted(
```

Comment:
Do we need to worry about spicy window in the runs/bulk action table?

<img width="868" alt="Screenshot 2024-08-06 at 10 01 07 AM" src="https://github.com/user-attachments/assets/98a3ecf8-08e2-41c4-936c-3990a73bf2f3">

Reply:
technically the problem can occur but I wouldn't worry about it in this context